internal/integration_test/vs/runtime.go

```diff
@@ -86,6 +89,16 @@ func (r *wazeroRuntime) Compile(ctx context.Context, cfg *RuntimeConfig) (err er
 			ExportFunction("log", r.log).Compile(ctx, wazero.NewCompileConfig()); err != nil {
 			return err
 		}
+	} else if cfg.EnvFReturnValue != 0 {
+		if r.env, err = r.runtime.NewModuleBuilder("env").
+			ExportFunction("f",
+				// Note: accepting (context.Context, api.Module) is the slowest type of host function with wazero.
+				func(context.Context, api.Module, uint64) uint64 {
```

Comment:
I think probably perf is same as we don't optimize much for the new `UsesMemory` yet, or we do and it doesn't matter :)